### PR TITLE
Add tabs support for scripter plugin

### PR DESCRIPTION
**Pull request by chefe**
Add support to the scripter plugin to define tabs while creating a new paragraph style.

## Example usage
```py
createParagraphStyle('MyNewStyle', tabs='1|8|. , 2|20')
```

### CODE REVIEW

**Comment by aoloe** [comment]
the idea to add tabs is (very) good... i'm not sure about the way for adding the values, though...

if i understand it correctly, you're defining it as a string that will be parsed and in the example above you're using:

- `,` as a _tab_ separator
- `|` as a field separator
- `1` and `2` as magic values for the type.

i would suggest you to use tuples and constants instead of the single string... so that somebody can read the definition without having to read the docs : - )

my proposition would be: 

```py
tabs = ((scribus.TAB_LEFT, 8, '.'), (scribus.TAB_CENTER, 20))
```

of course, having names for each field would be even better... but it looks already much more readable in this way...

what do you think?

**Comment by aoloe** [comment]
you might also want to first set the position, then the type...

```
tabs = (1, 2, 3)
```

would then define left tabs at position 1, 2, and 3... since the left tabs are most common ones...

**Comment by chefe** [comment]
@aoloe Thanks for your feedback. I will let you know when I have another version with an improved syntax to define tabs.

**Comment by aoloe** [comment]
do you need help?

**Comment by chefe** [comment]
I have a new implementation, which is nearly matches your proposal. The outer object is now a list `[]` and a tab is defined with a tuple syntax `()`. If tab position and filling char are omitted, a tab can be defined as with a single float value for the position. 

```
tabs = [(8.5, scribus.TAB_LEFT, '.'), (20, scribus.TAB_CENTER), 15.7]
```

Helper constants for the alignment are added in 549e308.

@aoloe Is this version better? What do you think?

File: scribus/plugins/scriptplugin/scriptplugin.cpp [hunk 1]
```diff
@@ -784,6 +785,12 @@ void initscribus(ScribusMainWindow *mainWin)
 	PyDict_SetItemString(d, const_cast<char*>("PAGE_2"), Py_BuildValue(const_cast<char*>("i"), 1));
 	PyDict_SetItemString(d, const_cast<char*>("PAGE_3"), Py_BuildValue(const_cast<char*>("i"), 2));
 	PyDict_SetItemString(d, const_cast<char*>("PAGE_4"), Py_BuildValue(const_cast<char*>("i"), 3));
+	// tab alignment
+	PyDict_SetItemString(d, const_cast<char*>("TAB_LEFT"), Py_BuildValue(const_cast<char*>("i"), ParagraphStyle::AlignmentType::LeftAligned));
```
Comment (aoloe) [inline scribus/plugins/scriptplugin/scriptplugin.cpp]: those are not the right values for the alignment constants.

i could not find any in `styles/paragraphstyle.h`... scribus seems to be using plain integers... (but you should defined constants for the scripter... of course : - )

the values are:

- Left
- Right
- Period
- Comma
- Center

which are probably integers from 0 to 4...

**Comment by aoloe** [comment]
i did not try to compile the code, but the code looks ok to me!
(we should move the pull requests to gitlab, then i would have downloaded the appimage and tested : - )

... the only issue are the the _wrong_ type constants. but this  should be easy to fix (see the inline comment to the latest commit)...

good work chef(e)!

**Comment by aoloe** [comment]
i guess it's time to re-post the patch in the official scribus bug tracker!

**Comment by chefe** [comment]
New version of the patch is attached to the official bug tracker. see issue [15800](https://bugs.scribus.net/view.php?id=15800#c46711).

**Comment by aoloe** [comment]
merged in with ccb3b871 .

thanks @chefe for this patch!

**Pull request closed without merging**